### PR TITLE
Wait for Suggestions Database Initialization

### LINKED ISSUE #978: Getting Suggestion Database results in "Suggestions database error"
### General Summary
I run the integration test in IDE and got error at first getSuggestionDatabase call. The logs in Engine suggest that this is some internal error.

### Steps to Reproduce
Run Language Server with command `sbt "runner/run --server --root-id 6f7d58dd-8ee8-44cf-9ab7-9f0454033641 --path $HOME/ensotmp --rpc-port 30616"` 
call `search/getSuggestionsDatabase`

### Expected Result
Some database

### Actual Result
"Suggestions database error", and the following exception is reported in console:
```
[info] 10:54:21.606 ERROR [scala-execution-context-global-55] o.e.l.r.s.GetSuggestionsDatabaseHandler GetSuggestionsDatabase error
[info] org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such table: suggestions)
[info] 	at org.sqlite.core.DB.newSQLException(DB.java:1010)
[info] 	at org.sqlite.core.DB.newSQLException(DB.java:1022)
[info] 	at org.sqlite.core.DB.throwex(DB.java:987)
[info] 	at org.sqlite.core.NativeDB.prepare_utf8(Native Method)
[info] 	at org.sqlite.core.NativeDB.prepare(NativeDB.java:134)
[info] 	at org.sqlite.core.DB.prepare(DB.java:264)
[info] 	at org.sqlite.core.CorePreparedStatement.<init>(CorePreparedStatement.java:45)
[info] 	at org.sqlite.jdbc3.JDBC3PreparedStatement.<init>(JDBC3PreparedStatement.java:30)
[info] 	at org.sqlite.jdbc4.JDBC4PreparedStatement.<init>(JDBC4PreparedStatement.java:19)
[info] 	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:35)
[info] 	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:241)
[info] 	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:205)
[info] 	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:311)
[info] 	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
[info] 	at slick.jdbc.JdbcBackend$SessionDef.prepareStatement(JdbcBackend.scala:375)
[info] 	at slick.jdbc.JdbcBackend$SessionDef.prepareStatement$(JdbcBackend.scala:365)
[info] 	at slick.jdbc.JdbcBackend$BaseSession.prepareStatement(JdbcBackend.scala:489)
[info] 	at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:33)
[info] 	at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:22)
[info] 	at slick.jdbc.Invoker.foreach(Invoker.scala:47)
[info] 	at slick.jdbc.Invoker.foreach$(Invoker.scala:46)
[info] 	at slick.jdbc.StatementInvoker.foreach(StatementInvoker.scala:16)
[info] 	at slick.jdbc.StreamingInvokerAction.run(StreamingInvokerAction.scala:22)
[info] 	at slick.jdbc.StreamingInvokerAction.run$(StreamingInvokerAction.scala:20)
[info] 	at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$2.run(JdbcActionComponent.scala:214)
[info] 	at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$2.run(JdbcActionComponent.scala:214)
[info] 	at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)
[info] 	at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)
[info] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info] 	at java.base/java.lang.Thread.run(Thread.java:834)
```

### Enso Version
commit `6ba038c800485910a925a9328119f7446000cf57`

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #978 

Add initialization mechanism to the Language Server.

### Important Notes

- add: InitializedEvent
- update: SuggestionsHandler to wait until the database is initialized
- refactor: move the search request logic to the `search` package

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Review by iamrecursion — APPROVED** [review]

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala [hunk 1]
```diff
@@ -35,16 +35,29 @@ class LanguageServerComponent(config: LanguageServerConfig)
...
+    val bindJsonServer =
+      for {
+        binding <- module.jsonRpcServer.bind(config.interface, config.rpcPort)
+        _       <- Future { logger.debug("Json RPC server initialized") }
+      } yield binding
+    val bindBinaryServer =
+      for {
+        binding <- module.binaryServer.bind(config.interface, config.dataPort)
+        _       <- Future { logger.debug("Binary server initialized") }
+      } yield binding
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala:53]: The module's initialization and socket binding will run concurrently. I'm not sure if it is the behavior you wanted to implement.

File: engine/language-server/src/main/scala/org/enso/languageserver/event/InitializedEvent.scala [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.enso.languageserver.event
+
+/** Event about the initialization of the language server component. */
+sealed trait InitializedEvent extends Event
+
+object InitializedEvent {
+
+  case object SuggestionsRepo  extends InitializedEvent
+  case object FileVersionsRepo extends InitializedEvent
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/event/InitializedEvent.scala]: Please rename it to `SuggestionRepoInitialized` and `FileVersionRepoInitialized`. Your names suggest that it is a repo object.

File: engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala [hunk 3]
```diff
@@ -239,6 +247,20 @@ final class SuggestionsHandler(
       context.become(initialized(name, clients))
   }
 
+  /**
+    * Transition the initialization process.
+    *
+    * @param state current initialization state
+    */
+  private def becomeInitialized(
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala]: If the initialization may not succeed, the method shouldn't have a name `becomeInitialized`. The better name would be `tryInitialize`.

**Review by lolczak — CHANGES_REQUESTED** [review]

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala (hunk 1 above)
Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala:53]: They are independent actions, and I indeed wanted to run them in parallel. This way it shaves ~2 seconds off the startup time on my machine.

**Review by lolczak — APPROVED** [review]

**Pull request merged**